src/Riok.Mapperly/MapperNode.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Riok.Mapperly;
+
+public readonly struct MapperNode : IEquatable<MapperNode>
+{
+    public MapperNode(CompilationUnitSyntax body, string fileName)
+    {
+        Body = body;
+        FileName = fileName;
+    }
+
+    public CompilationUnitSyntax Body { get; }
+    public string FileName { get; }
+
+    public bool Equals(MapperNode other) => Body.IsEquivalentTo(other.Body) && FileName == other.FileName;
+
+    public override bool Equals(object? obj) => obj is MapperNode other && Equals(other);
+
+    public override int GetHashCode()
+    {
+        unchecked
+        {
+            return (Body.GetHashCode() * 397) ^ FileName.GetHashCode();
```

Comment:
Couldn't this return another hash code for entities which are actually equal? (If the `Body` is another instance which is equal to each other?)

Reply:
Good question, it might.

I don't think incremental generators use the objects hash code or `SyntaxNode` has some custom `GetHashCode` method. Otherwise the tests would never pass and it wouldn't be possible to cache `SyntaxNode`, defeating the purpose of the incremental source generator.

Reply:
IMO not using the hash code is an implementation detail of how Roslyn calls the source generators which Mapperly should not rely on. IMO the fix here is also pretty simple:
* Instead of using the `Body.GetHashCode()` use a simpler hash code, eg. `Body.Members.Count.GetHashCode()`
* Remove the hash code part of the body completely